core/src/soundfont/mod.rs

```diff
@@ -198,25 +204,72 @@ impl SampleSoundfont {
             let params = sample_cache_from_region_params(&region);
             let envelope = envelope_descriptor_from_region_params(&region);
 
-            let speed_mult =
-                get_speed_mult_from_keys(region.key, region.pitch_keycenter.unwrap_or(region.key));
-
-            let envelope_params = unique_envelope_params
-                .iter()
-                .find(|e| e.0 == envelope)
-                .unwrap()
-                .1
-                .clone();
-
-            let spawner_params = Arc::new(SampleVoiceSpawnerParams {
-                envelope: envelope_params,
-                speed_mult,
-                sample: samples[&params].clone(),
-            });
-
-            for vel in region.lovel..=region.hivel {
-                let index = key_vel_to_index(region.key, vel);
-                spawner_params_list[index] = Some(spawner_params.clone());
+            if region.lokey.is_none() && region.hikey.is_none() {
```

Comment:
As I suggested in DMs, it might be easier to just use ranges in region params rather than hi/lo values, so these if statements aren't necessary.

Reply:
Yes, I'll change this.